spec/formatting.md

```diff
@@ -202,108 +264,63 @@ the following steps are taken:
-> currently implementation-dependent. Supposing that
-> the external input variable `n` is bound to the string `"1"`,
-> and that the implementation formats to a string,
-> the formatted result of the following message:
->
-> ```
-> .input {$n :number minimumIntegerDigits=3}
-> .local $n1 = {$n :number maximumFractionDigits=3}
-> {{What is the value of: {$n1}}}
-> ```
->
-> is currently implementation-dependent.
-> Depending on whether the options are preserved
-> between the resolution of the first `:number` _function_
-> and the resolution of the second `:number` _function_,
-> a conformant implementation
-> could produce either "001.000" or "1.000"
->
-> Each function **specification** MAY have
-> its own rules to preserve some options in the returned structure
-> and discard others. 
-> In instances where a function specification does not determine whether an option is preserved or discarded,
-> each function **implementation** of that specification MAY have
-> its own rules to preserve some options in the returned structure
-> and discard others. 
->
-
-> [!NOTE]
-> During the Technical Preview,
-> feedback on how the registry describes
-> the flow of _resolved values_ and _options_
-> from one _function_ to another,
-> and on what requirements this specification should impose,
-> is highly desired.
-
    An implementation MAY pass additional arguments to the function,
    as long as reasonable precautions are taken to keep the function interface
    simple and minimal, and avoid introducing potential security vulnerabilities.
 
-   An implementation MAY define its own functions.
-   An implementation MAY allow custom functions to be defined by users.
-
-   Function access to the _formatting context_ MUST be minimal and read-only,
-   and execution time SHOULD be limited.
-   
-   Implementation-defined _functions_ SHOULD use an implementation-defined _namespace_.
-
 5. If the call succeeds,
    resolve the value of the _expression_ as the result of that function call.
 
    If the call fails or does not return a valid value,
    emit the appropriate _Message Function Error_ for the failure.
 
-   Implementations MAY provide a mechanism for the _function_ to provide
+   Implementations MAY provide a mechanism for the _function handler_ to provide
    additional detail about internal failures.
    Specifically, if the cause of the failure was that the datatype, value, or format of the
    _operand_ did not match that expected by the _function_,
    the _function_ might cause a _Bad Operand_ error to be emitted.
   
-   In all failure cases, use the _fallback value_ for the _expression_ as the resolved value.
+   In all failure cases, use the _fallback value_ for the _expression_ as its _resolved value_.
+
+#### Function Handler
+
+To resolve a _function_,
+a **_<dfn>function handler</dfn>_** is required.
+This is an implementation-defined process such as a function or method
+which accepts a set of arguments and returns a _resolved value_.
```

Comment:
Same comment as before about `dfn`. The definition is in the second sentence. Lead with that and then require it.